src/Storages/IStorage.h

```diff
@@ -132,6 +132,9 @@ class IStorage : public std::enable_shared_from_this<IStorage>, public TypePromo
     /// Returns true if the storage supports insert queries with the PARTITION BY section.
     virtual bool supportsPartitionBy() const { return false; }
 
+    /// Returns true if the storage supports queries with the TTL section.
+    virtual bool supportsTTL() const { return false; }
```

Comment:
Let's also add a comment in https://github.com/ClickHouse/ClickHouse/blob/98a94eae7d63fcfb0ec2d7ac62096adc09989a61/src/Storages/StorageFactory.h#L62

Reply:
Yes, both need to be consistent.